### PR TITLE
Pin Github actions, group dependabot

### PR DESCRIPTION
**Pull request by jaimergp**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

xref https://github.com/conda-forge/infrastructure/pull/1

### CODE REVIEW

File: .github/workflows/autoupdate.yml [hunk 1]
```diff
@@ -1,23 +1,23 @@
...
- schedule:
-   - cron: "0 */6 * * *"
+  schedule:
+    - cron: "0 */6 * * *"
 jobs:
   createPullRequest:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v4
+      - uses: actions/checkout@0ad4b8fadaa221de15dcec353f45205ec38ea70b # v4
```
Comment (xhochy) [inline .github/workflows/autoupdate.yml:9]: Please don't add comments. Dependabot doesn't update them and this they become stale

Comment (jaimergp) [reply .github/workflows/autoupdate.yml:9]: I think it does? See this PR, for example: https://github.com/conda/conda/pull/13892

Comment (xhochy) [reply .github/workflows/autoupdate.yml:9]: Oh, that's new. We have a lot of these stale comments lingering in our internal repos 🤔

**Review by xhochy — APPROVED** [review]

File: .github/workflows/autoupdate.yml [hunk 2]
```diff
@@ -1,23 +1,23 @@
...
+      - uses: conda-incubator/setup-miniconda@a4260408e20b96e80095f42ff7f1a15b27dd94ca # v3.0.4
         with:
           miniforge-variant: Miniforge3
           environment-file: .github/actions/autoupdate/environment.yml
       - run: python .github/actions/autoupdate/update.py
       - name: Create Pull Request
         id: cpr
         # This is the v3 tag but for security purposes we pin to the exact commit.
-        uses: peter-evans/create-pull-request@6d6857d36972b65feb161a90e484f2984215f83e
+        uses: peter-evans/create-pull-request@6d6857d36972b65feb161a90e484f2984215f83e # 6d6857d36972b65feb161a90e484f2984215f83e
```
Comment (jaimergp) [inline .github/workflows/autoupdate.yml:20]:
```suggestion
        uses: peter-evans/create-pull-request@6d6857d36972b65feb161a90e484f2984215f83e # v6.0.5
```

**Pull request merged**